FRW-1442 Migration guide for Country, Currencty, Locale modules (#1563)
spryker/spryker-docs: _includes/pbc/all/upgrade-modules/upgrade-the-locale-module.md

**Application integration**

`src/Pyz/Zed/Application/ApplicationDependencyProvider.php`

```php
<?php

namespace Pyz\Zed\Application;

use Spryker\Zed\Application\ApplicationDependencyProvider as SprykerApplicationDependencyProvider;
use Spryker\Zed\Locale\Communication\Plugin\Application\LocaleBackendGatewayApplicationPlugin;

class ApplicationDependencyProvider extends SprykerApplicationDependencyProvider
{
    ...

    /**
     * @return \Spryker\Shared\ApplicationExtension\Dependency\Plugin\ApplicationPluginInterface[]
     */
    protected function getBackendGatewayApplicationPlugins(): array
    {
        return [
            ...
            new LocaleBackendGatewayApplicationPlugin(),
            ...
        ];
    }

    ...
}
```

**Integrate console commands**

`src/Pyz/Zed/Console/ConsoleDependencyProvider.php`

```php
<?php

namespace Pyz\Zed\Console;

use Spryker\Zed\DataImport\Communication\Console\DataImportConsole;
use Spryker\Zed\Console\ConsoleDependencyProvider as SprykerConsoleDependencyProvider;
use Spryker\Zed\Locale\Communication\Plugin\Console\LocaleApplicationPlugin;
use Spryker\Zed\LocaleDataImport\LocaleDataImportConfig;

> use Spryker\Zed\Locale\Communication\Plugin\Console\LocaleApplicationPlugin;

Reviewer: Locale application plugin is duplicated.

class ConsoleDependencyProvider extends SprykerConsoleDependencyProvider 
{
    /**
     * @var string
     */
    protected const COMMAND_SEPARATOR = ':';
    
    /**
     * @param \Spryker\Zed\Kernel\Container $container
     *
     * @return array<\Symfony\Component\Console\Command\Command>
     */
    protected function getConsoleCommands(Container $container): array
    {
        return [
            ...
            new DataImportConsole(DataImportConsole::DEFAULT_NAME . static::COMMAND_SEPARATOR . LocaleDataImportConfig::IMPORT_TYPE_LOCALE_STORE),
            new DataImportConsole(DataImportConsole::DEFAULT_NAME . static::COMMAND_SEPARATOR . LocaleDataImportConfig::IMPORT_TYPE_DEFAULT_LOCALE_STORE),
            ...        
        ];
    }
    
    /**
     * @param \Spryker\Zed\Kernel\Container $container
     *
     * @return array<\Spryker\Shared\ApplicationExtension\Dependency\Plugin\ApplicationPluginInterface>
     */

>     public function getApplicationPlugins(Container $container): array

Reviewer: `Container` should be added to `use` section.

This document describes how to upgrade the Locale module.

## Upgrading from version 3.* to version 4.0.0

> In this new version of the `Locale` module, we have added support configuration locale for each store in database.

Reviewer 2: ```suggestion
In this new 4.0.0 version of the `Locale` module, we have added support configuration locale for each store in the database.
```

> With the `Locale` module version 4 we have added the `spy_locale_store` database table to persist stores-locales in Zed.

Reviewer 2: ```suggestion
With the new 4.0.0 version of the `Locale` module, we have also added the `spy_locale_store` database table to persist stores-locales in Zed.
```

> Also added column `fk_locale` into  `spy_store` for save default locale per store. 

Reviewer 2: ```suggestion
We also added the column `fk_locale` into  `spy_store`, to allow saving a default locale per store. 
```

You can find more details about the changes on the [Locale module](https://github.com/spryker/locale/releases) release page.

> *Estimated migration time: 5 min*

Reviewer 2: ```suggestion
*Estimated migration time: 5 minutes*
```
